### DISCUSSION TITLE
Benefit of GramUp over directly using Telegram

### DISCUSSION THREAD

**Discussion by khemrajrathore** [rohittp0/GramUp#46] [Q&A]
@rohittp0 Nice project appreciate the idea and effort, But I have a question.

Telegram provides a 2GB per file limit and no restrictions on the total storage used. Can I not use the telegram app directly to send the documents to a chat and all my documents will be there. Why would I use a third-party app to use the functionality which Telegram already provides? 

Enlighten me, If I haven't understood the use-case or limitation on either side.

**Comment by rohittp0** [comment]
Good question, I should have clarified this in the README. So the point of Gramup is to provide a better UI and some additional functionality over Telegram. Gramup provides two major features ( other than the UI ):
- Track and sync local files - Using gramup you don't have to manually upload every time there is a file change or a new file is added to a tracked folder 
- Preserve folder structure - When you are uploading to Telegram all Telegram sees are individual files and say if there is a file `folder-a/file` and another file `folder-b/file`,  once uploaded to Telegram the fact that these were files in different folders is lost.

Telegram's file upload feature is majorly for file sharing, and Gramup adapts it as a backup solution.
